formidable/forms/__init__.py

```diff
@@ -50,10 +50,27 @@ def __init__(self, *args, **kwargs):
         super(BaseDynamicForm, self).__init__(*args, **kwargs)
         self._bound_fields_cache = FormidableBoundFieldCache()
 
+    def rule_has_empty_fields(self, rule, data):
```

Comment:
> sometimes you don't want the rule to be aware of the form/fields and sometimes you want :)

Actually, you don't use any "self" in the current method. It's a sign for me this method does not belongs to the form but to the rules itself. 

> sometimes you don't want the rule to be aware of the form/fields and sometimes you want :)

I didn't see any relation with what I'm saying, your method check "data" came from the form, not data in the form/fields itself.